workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/apps/log_viewer.py

```diff
@@ -0,0 +1,85 @@
+import sys
+import streamlit as st
+import vcm.fv3.logs
+import fsspec
+from toolz.curried import map, compose, reduce
+import pandas as pd
+import plotly.express as px
+import subprocess
+
+
+# https://davidmathlogic.com/colorblind/#%23000000-%23E69F00-%2356B4E9-%23009E73-%23F0E442-%230072B2-%23D55E00-%23CC79A7
+wong_palette = [
+    "#000000",
+    "#E69F00",
+    "#56B4E9",
+    "#009E73",
+    "#F0E442",
+    "#0072B2",
+    "#D55E00",
+    "#CC79A7",
+]
+
+
+def concat_logs(a, b):
+    return vcm.fv3.logs.FV3Log(
+        dates=a.dates + b.dates,
+        totals={key: a.totals[key] + b.totals[key] for key in b.totals},
+        ranges=None,
+    )
+
+
+def open_segmented_logs(url):
+    fs = fsspec.filesystem("gs")
+    logfiles = sorted(fs.glob(f"{url}/**/logs.txt"))
+    parsed = reduce(
+        concat_logs, map(compose(vcm.fv3.logs.loads, bytes.decode, fs.cat), logfiles)
+    )
+
+    df = pd.DataFrame({species: parsed.totals[species] for species in parsed.totals})
+    df.index = parsed.dates
+    return df
+
+
+def line_chart(df, url):
+    df = pd.melt(df.reset_index(), id_vars=["index"])
+    fig = px.line(df, x="index", y="value", color="variable", title=url).update_layout(
+        font=dict(size=10)
+    )
+    return st.plotly_chart(fig)
+
+
+def register_parser(subparsers):
+    parser = subparsers.add_parser(
+        "log-viewer", help="Webapp for viewing segmented run logs."
```

Comment:
to me, a 'log-viewer' is a tool to view or query logs, not make plots. IMO we'll probably want to extend this to the other text outputs, so maybe we should call it `statistic-viewer` or something? Or be a bit more specific in the help message.

Reply:
Logs is a somewhat established convention here. These statistics are parsed from the logs file using the functionality in `vcm.fv3.logs`. I will add some information to help.

Reply:
Also "view" is where all the plotting code is in this prognostic run.